### PR TITLE
Add kotd library

### PR DESCRIPTION
**Pull request by jtzhao**
Add "kernel of the day" library to help kotd testing

### CODE REVIEW

File: tests/kernel/install_kotd.pm [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+# Install kotd kernel
+sub kotd_install {
+    zypper_call("install --from kotd kernel-default", timeout => 1200);
+}
+
+# Reboot system and login
+sub kotd_reboot {
+    type_string("reboot\n");
+    wait_boot;
+    reset_consoles();
```
Comment (coolo) [inline tests/kernel/install_kotd.pm]: wait_boot is already doing that - and it even selects the root-console if you pass textmode => 1 to it.

Comment (jtzhao) [inline tests/kernel/install_kotd.pm]: Fixed.

**Comment by richiejp** [comment]
It looks like it should run perfectly, but a verification run would be nice. For example, you could write a test module which just calls these functions and then checks `uname -r` with a regex.

**Comment by jtzhao** [comment]
@richiejp can't find a working up-to-date openQA server. Every time I upgrade openQA server, it breaks.

**Comment by richiejp** [comment]
I have not had any problems upgrading OpenQA with zypper. If you are using the os-autoinst source then you may need to recompile the C++ components. You can ask for help fixing any problems on IRC or on the mailing list.

**Comment by foursixnine** [comment]
@jtzhao Can you define "the server breaks"? What's your current setup? (As @richiejp mentioned, you can ask for help on irc)

**Comment by richiejp** [comment]
It seems that @pevik may have the same issue. It might just be because some shared libraries are outdated and you need to restart all the services (just restart the whole system).

**Comment by jtzhao** [comment]
Removed kotd_reboot function because I can't get it working. We can reboot inside the actual test module though.
Test result: http://147.2.207.102/tests/630#step/test/21

**Comment by okurz** [comment]
please [squash commits](https://robots.thoughtbot.com/git-interactive-rebase-squash-amend-rewriting-history)

verification looks ok but still the library is not used anywhere in the test distribution so I do not see the purpose of this module (yet).

**Comment by jtzhao** [comment]
@okurz https://progress.opensuse.org/issues/15042

**Comment by jtzhao** [comment]
I've converted this module into a test and added it into products/sle/main.pm. Now during ltp and regression testing, if INSTALL_KOTD is set(e.g. SLE12-SP3), kotd kernel will be installed ahead of other steps.
Test result: http://147.2.207.102/tests/648

**Comment by richiejp** [comment]
Excellent.

**Pull request merged**